modules/internal/ChapelArray.chpl

```diff
@@ -1378,6 +1392,24 @@ module ChapelArray {
       return _newArray(x);
     }
 
+    pragma "no doc"
+    pragma "no copy return"
+    proc buildArrayWith(type eltType, data: c_ptr(eltType), allocSize) {
+      if eltType == void {
+        compilerError("array element type cannot be void");
+      }
+      var x = _value.dsiBuildArrayWith(eltType, data, allocSize);
+
+      pragma "dont disable remote value forwarding"
+      proc help() {
+        _value.add_arr(x);
+      }
+      help();
+
+      chpl_incRefCountsForDomainsInArrayEltTypes(x, x.eltType);
+
+      return _newArray(x);
+    }
```

Comment:
I think you're trying to ask if buildArrayWith or _getArray make a better user-facing interface? I wouldn't call it _getArray but it makes sense to me that you wouldn't need to create the domain before creating the array, in cases like this, since the domain is always 0..#size anyway.